geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/fs/storage/parquet/io/SimpleFeatureParquetSchema.scala

```diff
@@ -56,16 +74,64 @@ case class SimpleFeatureParquetSchema(sft: SimpleFeatureType, schema: MessageTyp
 
 object SimpleFeatureParquetSchema {
 
+  import StringSerialization.alphaNumericSafeString
+  import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+
   import scala.collection.JavaConverters._
 
   val FeatureIdField = "__fid__"
 
   val SchemaVersionKey = "geomesa.parquet.version"
 
-  val CurrentSchemaVersion = 1
+  val CurrentSchemaVersion = 2
 
-  val GeometryColumnX = "x"
-  val GeometryColumnY = "y"
+  val Encoding = "WKB"
+  val GeoParquetSchemaKey = "geo"
+
+  /**
+   * See https://geoparquet.org/releases/v1.0.0/schema.json
+   *
+   * @param sft simple feature type
+   * @return
+   */
+  def geoParquetMetadata(sft: SimpleFeatureType): String = {
+    val geomField = sft.getGeomField
+
+    // If the sft has no geometry field, then omit the GeoParquet metadata entirely
+    if (geomField == null) {
+      ""
+    } else {
+      val primaryColumn = alphaNumericSafeString(geomField)
+      val columns = sft.getAttributeDescriptors.asScala.collect { case g: GeometryDescriptor => geoParquetMetadata(g) }.mkString(",")
+
+      s"""{"version":"1.1.0-dev","primary_column":"$primaryColumn","columns":$columns"""
```

Comment:
I think version here needs to be `1.0.0` according the the json-schema

Reply:
colums here doesn't seem quite right - it should end up looking like:
```
columns: {
  "geomA": {
    "encoding": "WKB",
    "geometry_types": []
  },
  "geomB": {
    "encoding": "WKB",
    "geometry_types": [ "Point" ],
    "bbox": [0, 0, 1, 1]
  },
}
```
but i think the way you have it now it won't work for multiple geometries.

Reply:
since this json is getting a little complicated, it might be good to model it using case classes and then use something like json4s to convert it to json.